src/api/libcellml/types.h

```diff
@@ -31,6 +32,14 @@ class Units; /**< Forward declaration of Units class. */
 typedef std::shared_ptr<Units> UnitsPtr; /**< Type definition for shared units pointer. */
 class Variable; /**< Forward declaration of Variable class. */
 typedef std::shared_ptr<Variable> VariablePtr; /**< Type definition for shared variable pointer. */
+class XmlNode; /**< Forward declaration of XmlNode class. */
+typedef std::shared_ptr<XmlNode> XmlNodePtr; /**< Type definition for shared XML node pointer. */
```

Comment:
They are part of the interface albeit the private interface nevertheless we need to forward declare these objects.  It's an improvement over including non-API headers in the API header files.